### ISSUE TITLE
[Scenario Analysis]: Sharing user level flatpaks between systems via systemd-homed over different architectures

### ISSUE THREAD

**Issue by MateusRodCosta** [flatpak/flatpak#6044]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a feature request that matches the one I want to file, without success.

### Suggestion

This isn't as much a feature request but instead a scenario analysys that I thought about and mentioned on Flatpak Matrix room. This is mostly due to systemd-home and Fedora Asahi on Apple Sillicon.

Essentially, if you install flatpaks in user mode (`flatpak --user`), the flatpaks are installed inside the user folder. systemd-homed allows you to have portable home folders, so one possibility would be to share the user home between two systems with either same architecture or different ones.

If you were to share between system of same architecture (such as two x86_64 systems), I would assume that as long as flatpak is properly setup there would be no issues. The same flatpaks are shared and there are virtually no changes.

My question really is about what if sharing between different architectures? (such as original x86_64 and a ARM64 system)

I know that if a system tries to run a binary for a different architecture it will usually fallback to QEMU with low performance, but I know that FEX exists for ARM to run x86 with more performance.

That would be if just using the same x86_64 flatpaks, I would assume the user could download the ARM64 variants either manually or flatpak could detect the need based on the host architecture. That assumes, of course, good deduplication between the same flatpak in different arches.

Somebody mentioned also that flatpak isn't smart enough to run the correct app based on arch, it only really runs the one set with "make-current".

So, to sum up, what would be needed for the cross-platform sharing of a flatpak installation to work?

**Comment by kevATin** [comment]
This is something that I'm also very interested in.
But what I had in mind was using your flatpaks on both your desktop and on your Linux phone (or a Raspberry Pi or similar). 
Ideally developers should be able to bundle any supported binaries of their app together into one flatpak and upon starting it the system should automatically pick the correct one.
Flatpaks are already not exactly known for their lightweightness so they are unlikely to be used on very storage-constrained devices, so we might as well make use of that space.
This may not be feasible for every app, but I see little reason why something like a simple calculator flatpak couldn't support x86, x64, arm32, arm64, riscv32, riscv64, etc.
The "Available Architectures" on Flathub would have to be changed to reflect whether an app supports the individual architectures or one flatpak with multiple of them.
